### PR TITLE
Few minor changes in some feature extractors to automatically use the dependency view 

### PR DESCRIPTION
**Pull request by danyaljj**
Following the discussion in #106

### CODE REVIEW

File: edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ChunkPathPattern.java [hunk 1]
```diff
@@ -56,56 +53,55 @@ public ChunkPathPattern(String viewName) {
...
-        for (Constituent constituent : constituents) {
-            sb.append(constituent.getLabel()).append("-");
-        }
-
-        if (constituents.size() <= 1)
-            sb.append("<empty>");
-
+        List<Relation> incomingRelation = c.getIncomingRelations();
         Set<Feature> features = new LinkedHashSet<>();
+        if(incomingRelation.size() > 0) {
```
Comment (danyaljj) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ChunkPathPattern.java]: The only change here is adding this `if` block. The rest is the same.

File: edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java [hunk 2]
```diff
@@ -44,47 +44,46 @@ public ClauseFeatureExtractor(String pseudoParseViewName, String clauseViewName)
...
-            Constituent argParent = TreeView.getParent(arg);
-            Constituent predParent = TreeView.getParent(pred);
-            if (argParent == ca && predParent == ca)
-                clauseRelativePosition = "S";
-            else if (argParent == ca || arg == ca)
-                clauseRelativePosition = "A";
-            else if (predParent == ca || pred == ca)
-                clauseRelativePosition = "B";
-            else
+        if(incomingRelations.size() > 0) {
```
Comment (danyaljj) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/ClauseFeatureExtractor.java]: The only change here is this `if` block. The rest is the same.

File: edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/SyntacticFrame.java [hunk 3]
```diff
@@ -45,59 +47,59 @@ public SyntacticFrame(String parseViewName) {
 
     @Override
     public Set<Feature> getFeatures(Constituent c) throws EdisonException {
+        Set<Feature> features = new LinkedHashSet<>();
+        List<Relation> incomingRelations = c.getIncomingRelations();
+
+        if(incomingRelations.size() > 0) {
```
Comment (danyaljj) [inline edison/src/main/java/edu/illinois/cs/cogcomp/edison/features/factory/SyntacticFrame.java]: The only change here is this `if` block. The rest is the same.

**Comment by christos-c** [comment]
@danyaljj am I missing something here? How are the changes here equivalent to what you suggested in #106?

**Comment by danyaljj** [comment]
In a convoluted way! 😆 😆 
So here are two major the changes: 
 - Changed the instances of `c.getIncomingRelations().get(0)`, to `incomingRelations = c.getIncomingRelations(); if(incomingRelations.size > 0)  .... `. 
 - Feature extractors which work (only) with dependency parsing, given `Constituent c`, I first find `Constituent cEquivalentInParseView = parse.getConstituentsCoveringToken(c.getStartSpan()).get(0);` and continue the rest with `cEquivalentInParseView` (instead of `c`). 

Let me know if it is still unclear.

**Pull request merged**